processing/src/test/java/io/druid/segment/data/IncrementalIndexTest.java

```diff
@@ -339,8 +339,12 @@ public void run()
                           new LinkedList<Result<TimeseriesResultValue>>()
                       )
                       ) {
-                    if (someoneRan.get()) {
-                      Assert.assertTrue(result.getValue().getDoubleMetric("doubleSumResult0") > 0);
+                    final Integer ranCount = someoneRan.get();
+                    if (ranCount > 0) {
+                      final Double sumResult = result.getValue().getDoubleMetric("doubleSumResult0");
+                      // Eventually consistent, but should be somewhere in that range
+                      // Actual result is validated after all writes are guaranteed done.
+                      Assert.assertTrue(String.format("%d >= %g >= 0 violated", ranCount, sumResult), sumResult >= 0 && sumResult <= ranCount);
```

Comment:
Yes, the bugs previously were in the logic itself causing index out of bounds error, rather than in the `Assert` code. The `Assert` stuff is really just a sanity check.